libbindgen/tests/headers/constructors.hpp

```diff
@@ -0,0 +1,13 @@
+
+class TestOverload {
+  // This one shouldnt' be generated.
+  TestOverload();
+public:
+  TestOverload(int);
+  TestOverload(double);
```

Comment:
Do we handle when the class already has a `new` method properly? Do we just treat the constructors or the `new` methods as overloads? Care to add a small test case?